### ISSUE TITLE
Immutability when consuming FGraph

### ISSUE THREAD

**Issue by HarryMcCarney** [fslaborg/Graphoscope#45]
Might be a matter of taste but when using FSX files I find immutability very helpful.

Should we update operations like removeNode to create, update and return a copy of the passed in graph. 

This keeps the usage closer to a functional F# style and removes cognitive overhead of mutating objects in scope of interactive sessions.  Any thoughts?

**Comment by muehlhaus** [comment]
I completely agree. We should move all functions that mutate the graph in a module named 'InPlace' or add 'InPLace' as a suffix to the function name.

The function with classical naming should always return a copy.
